commands/container-registry/token/tokens.go

```diff
@@ -0,0 +1,184 @@
+	scope "github.com/ionos-cloud/ionosctl/v6/commands/container-registry/token/scopes"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/config"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/constants"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/core"
+	"github.com/ionos-cloud/ionosctl/v6/pkg/printer"
+	"github.com/ionos-cloud/ionosctl/v6/services/container-registry/resources"
+	ionoscloud "github.com/ionos-cloud/sdk-go-container-registry"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func TokenCmd() *core.Command {
+	tokenCmd := &core.Command{
+		Command: &cobra.Command{
+			Use:     "token",
+			Aliases: []string{"t", "tokens"},
+			Short:   "Registry Tokens Operations",
+			Long: "Manage container registries for storage of docker images and OCI compliant artifacts. " +
+				"This operation is restricted to contract owner, admin, and users with 'accessAndManageRegistries' and " +
+				"Share/Edit access permissions for the data center hosting the registry.",
+			TraverseChildren: true,
+		},
+	}
+
+	tokenCmd.Command.PersistentFlags().Bool(
+		constants.ArgNoHeaders, false, "When using text output, don't print headers",
+	)
+
+	tokenCmd.AddCommand(TokenListCmd())
+	tokenCmd.AddCommand(TokenPostCmd())
+	tokenCmd.AddCommand(TokenGetCmd())
+	tokenCmd.AddCommand(TokenDeleteCmd())
+	tokenCmd.AddCommand(TokenUpdateCmd())
+	tokenCmd.AddCommand(TokenReplaceCmd())
+	tokenCmd.AddCommand(scope.TokenScopesCmd())
+
+	return tokenCmd
+}
+
+func getTokenPrint(
+	resp *ionoscloud.APIResponse, c *core.CommandConfig, response *[]ionoscloud.TokenResponse,
+	post bool,
+) printer.Result {
+	r := printer.Result{}
+	cols, _ := c.Command.Command.Flags().GetStringSlice(constants.ArgCols)
+
+	if c != nil {
+		if resp != nil {
+			r.Resource = c.Resource
+			r.Verb = c.Verb
+			r.WaitForState = viper.GetBool(
+				core.GetFlagName(
+					c.NS, constants.ArgWaitForRequest,
+				),
+			) // this boolean is duplicated everywhere just to do an append of `& wait` to a verbose message
+		}
+		if response != nil {
+			if !post {
+				defaultHeaders := []string{"TokenId", "DisplayName", "ExpiryDate", "Status"}
+				r.OutputJSON = response
+				r.KeyValue = getTokensRows(response) // map header -> rows
+				r.Columns = printer.GetHeaders(
+					allCols, defaultHeaders, cols,
+				) // headers
+			} else {
+				r.OutputJSON = response
+				r.KeyValue = getTokensRows(response)
+				postHeaders := []string{"DisplayName", "ExpiryDate", "Status"} // map header -> rows
+				r.Columns = printer.GetHeaders(allCols, postHeaders, cols)     // headers
+			}
+		}
+	}
+	return r
+}
+
+type TokenPrint struct {
+	TokenId             string `json:"TokenId,omitempty"`
+	DisplayName         string `json:"DisplayName,omitempty"`
+	ExpiryDate          string `json:"ExpiryDate,omitempty"`
+	CredentialsUsername string `json:"CredentialsUsername,omitempty"`
+	CredentialsPassword string `json:"CredentialsPassword,omitempty"`
+	Status              string `json:"Status,omitempty"`
```

Comment:
would be nice to have a "scopes" column - but if it's not feasible to unmarshal it all into a column then it's ok to skip it